### PR TITLE
allow LIEF to fail when using patchelf

### PR DESCRIPTION
**Pull request by minrk**
avoids unconditional failure when LIEF raises, even when not used

### Description

related to #5175, this allows LIEF to fail just like patchelf.

Previously, even when specifying `method=patchelf`, LIEF was unconditionally run. This adds error handling matching patchelf, so errors in an unused method don't prevent proceeding with patchelf.

### CODE REVIEW

**Review by beeankha — CHANGES_REQUESTED** [review]
Hi @minrk , thank you so much for this PR! A few things are requested (or are open for discussion) prior to merge:

1. Is there any way that this code could be re-worked to use `log.warn` / `log.error` (with `log = utils.get_logger(__name__)`) for the error handling? I understand that you most likely wanted to remain consistent with the rest of the `mk_relative_linux` function, but we can take this opportunity to improve error handling patterns (if you feel inclined to do so!)

2. It would be great to see a test or two for this change; addtionally it might help to have more details in the PR description for steps on how to reproduce LIEF being unconditionally run even when `method=patchelf` was specified

3. Please add a news file with a brief description of this change

**Comment by minrk** [comment]
I switched it to log and added news. I'm afraid I don't know how to exercise this failure in a test.

File: news/5176-lief-fail [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Bug fixes
+
+* Error handling when LIEF fails is now consistent with patchelf
```
Comment (beeankha) [inline news/5176-lief-fail:3]:
```suggestion
* Error handling when `LIEF` fails is now consistent with `patchelf`. (#5176)
```

**Review by beeankha — DISMISSED** [review]

**Comment by jaimergp** [comment]
Hm, the modified function is now using two different methods to report errors and warnings to the user. This implies slightly different formats, and also different streams (logging will print to stderr and print to stdout, as currently configured). I'd rather stick to one of the methods so:

- Either change everything to logging (and move `logger` to the top-level module namespace, under the imports)
- Revert to print with the same format as the other messages. We would need to create an issue to tackle a general move to logging, if one doesn't exist yet

**Comment by beeankha** [comment]
Per Jaime's [comment,](https://github.com/conda/conda-build/pull/5176#issuecomment-2224985028) I reverted the logging changes; I also went ahead and [filed an issue](https://github.com/conda/conda-build/issues/5415) to change all error-reporting to logging.

**Review by dholth — APPROVED** [review]

**Pull request merged**